src/k8s-engine/lima.ts

```diff
@@ -1314,6 +1321,17 @@ ${ commands.join('\n') }
             }
           });
 
+        if (this.#currentContainerEngine !== ContainerEngine.MOBY) {
```

Comment:
```suggestion
        if (this.#currentContainerEngine === ContainerEngine.CONTAINERD) {
```
(Switch to === because that's the only situation we'd have kim, when upgrading from older versions.)

Also add a comment saying that we should _only_ change the stored pref if we're using containerd, so that if the user had previously run containerd (thus installing kim), switched to moby, and then upgraded, we'd still remove kim if they later switched back to containerd.

Reply:
Actually, there are no buildkitd artifacts to remove in that situation, but I've written it up in a longish comment.